### PR TITLE
Create new note - close #5

### LINKED ISSUE #5: Create A New Note
At the bottom of the `reminders.hbs` template, there should be a button that routes the user to `/reminders/new` where users can create a new reminder.

* The new reminder should have inputs for the title, date, and notes.
* The new reminder should have a button for submitting the new note.
* Pressing the button should submit a new reminder to the database and immediately displays in the reminders list.
**Hot Tip**: There are no tests set up for this user story. So, you’ll have to write an acceptance test.

### PR DESCRIPTION
**Pull request by ejwill04**
## Purpose

close #5 

At the bottom of the reminders.hbs template, there should be a button that routes the user to /reminders/new where users can create a new reminder.

- The new reminder should have inputs for the title, date, and notes.
- The new reminder should have a button for submitting the new note.
- Pressing the button should submit a new reminder to the database and immediately displays in the reminders list.

## Approach

We created a 'new' route and then created a 'new-note' template and component to render and store the new note.  We added acceptance and integration tests and left the basic unit test. 

### Learning

We used Ember docs and referenced code from the Grocery-list repo.

### Test coverage 

We wrote both integration and acceptance tests.

The integration tests are used to check for default values and then that the inputs can be updated.  The acceptance tests are used to verify that a new note can be added and rendered on the page with the correct data.  

### Follow-up tasks

- [Issue #6  (Don’t display notes if there are no notes)](https://github.com/turingschool-projects/1610-remember-6/issues/6)

#### Before
screen shot 2017-02-13 at 6 19 36 am

#### After
screen shot 2017-02-13 at 6 19 14 am

#### Tests
screen shot 2017-02-13 at 6 20 31 am

### CODE REVIEW

File: tests/acceptance/new-note-test.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import { test } from 'qunit';
+import moduleForAcceptance from 'remember/tests/helpers/module-for-acceptance'
+
+moduleForAcceptance('Acceptance | new note')
+
+  test('should add a new note on submit with valid input', function(assert) {
+    visit('reminders/new');
+    fillIn('.spec-input-title', 'Banana');
```
Comment (martensonbj) [inline tests/acceptance/new-note-test.js]: I see you're pulling code from the groceries app. Doesn't seem like content that a reminder would contain. I'm fine with you using other projects as an example but make sure you aren't copy/pasting.

File: tests/integration/components/new-note-test.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+test('it renders with default string properties', function(assert) {
+  this.render(hbs`{{new-note}}`);
+  assert.equal(this.$('.spec-input-title').text().trim(), '', 'title is empty string');
+  assert.equal(this.$('.spec-input-date').text().trim(), '', 'date is empty string');
+  assert.equal(this.$('.spec-textarea-notes').text().trim(), '', 'notes is empty string');
+});
+
+test('it renders an updated values', function(assert) {
+  this.set('title', 'bob');
```
Comment (martensonbj) [inline tests/integration/components/new-note-test.js:16]: 👍 Yay! Props for doing an integration test.

File: tests/unit/routes/reminders/new-test.js [hunk 3]
```diff
@@ -0,0 +1,11 @@
+import { moduleFor, test } from 'ember-qunit';
```
Comment (martensonbj) [inline tests/unit/routes/reminders/new-test.js]: Don't need this file

**Review by martensonbj — CHANGES_REQUESTED** [review]

**Comment by ejwill04** [comment]
#### Updated Test Results (Removed one passing unit test)
screen shot 2017-02-13 at 12 46 43 pm

File: tests/acceptance/new-note-test.js (hunk 1 above)
Comment (ejwill04) [reply tests/acceptance/new-note-test.js]: I updated the examples for the test to be more reminder-friendly

File: tests/unit/routes/reminders/new-test.js (hunk 3 above)
Comment (ejwill04) [reply tests/unit/routes/reminders/new-test.js]: It's deleted on the latest commit

File: tests/acceptance/new-note-test.js [hunk 4]
```diff
@@ -0,0 +1,17 @@
+import { test } from 'qunit';
+import moduleForAcceptance from 'remember/tests/helpers/module-for-acceptance'
+
+moduleForAcceptance('Acceptance | new note')
+
+  test('should add a new note on submit with valid input', function(assert) {
+    visit('reminders/new');
+    fillIn('.spec-input-title', 'Eat a Banana');
+    fillIn('.spec-input-date', 'Now');
+    fillIn('.spec-textarea-notes', 'Only eat the extra green ones');
```
Comment (martensonbj) [inline tests/acceptance/new-note-test.js:10]: Better :)

**Review by martensonbj — APPROVED** [review]

**Pull request merged**